src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,40 +1,116 @@
-import { useState } from 'react';
+import { FC, FormEvent, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (newMovie: Movie) => void;
+}
+
+export const NewMovie: FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imageUrl, setImageUrl] = useState('');
+  const [imbdUrl, setImbdUrl] = useState('');
+  const [imbdId, setImbdId] = useState('');
+
+  const isFormFull = title && imageUrl && imbdUrl && imbdId;
+
+  const handleReset = () => {
+    setTitle('');
+    setDescription('');
+    setImageUrl('');
+    setImbdUrl('');
+    setImbdId('');
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!isFormFull) {
+      return;
+    }
+
+    const newMovieToPush: Movie = {
+      title: title,
+      description: description,
+      imgUrl: imageUrl,
+      imdbUrl: imbdUrl,
+      imdbId: imbdId,
+    };
```

Comment:
You can shorten it by using shorthand property names, as the object keys and variable names are the same
```suggestion
    const newMovieToPush: Movie = {
      title,
      description,
      imgUrl,
      imdbUrl,
      imdbId,
    };
```